File: python/350.py
from collections import Counter
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        hashtable = Counter()
        result = []
        for num1 in nums1:
            hashtable[num1] += 1
        for num2 in nums2:
            if num2 in hashtable:
                if hashtable[num2]:
                    result.append(num2)
                    hashtable[num2] -= 1
        return result

File: cpp/350.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> hashmap;
        // unordered_set<int> set;
        vector<int> res;
        for (int num1: nums1) ++hashmap[num1];
        for (int num2: nums2)
            if (hashmap.find(num2) != hashmap.end()) 
                if (hashmap[num2]) {
                    res.push_back(num2);
                    --hashmap[num2];
                }
        return res;
    }
};
